### ISSUE TITLE
Unfair comparison with SimpleBaseline and SimpleBaseline-darkpose

### ISSUE THREAD

**Issue by HuangJunJie2017** [yangsenius/TransPose#1]
Nice pioneer~
But it is not fair to compare TransPose-R with the original SimpleBaseline-Res, as the original SimpleBaseline use a weaker augmentation strategy and a shorter training schedule.  This will lead to a large gap of 1.5 AP. 
I look forward to a fair comparison between them for reference.

**Comment by yangsenius** [comment]
Thanks for your interest!

We use the same data augmentation strategy (following the HRNet code) for all our models. The long training schedule is a necessary setting for training our transformer-based models. Thank your suggestion. We would consider controlling the data augmentation strategy to be the same when comparing with SimpleBaseline.

**Comment by HuangJunJie2017** [comment]
emmm, under this situation, an ablation study is appealing and necessary. This make me curious that do the transformer-based models perform very poor when using the aug-schedule configuration in the original SimpleBaseline?

**Comment by yangsenius** [comment]
Hi, @HuangJunJie2017 . We did not train our models using the aug-schedule configuration of the original SimpleBaseline.

We provide the convergence curves and [log](https://gist.github.com/yangsenius/f6aaad3ad2eab37f39afa4b39924f7a0#file-gistfile1-txt) of the mAP for TransPose-R-A4. The curves are as below:

![Screenshot from 2021-01-27 10-54-49](https://user-images.githubusercontent.com/28689106/105942616-8656d600-609a-11eb-87bc-9842896e8603.png)

We also use the same data augmentation strategy of TransPose-R, 
```yaml
DATASET:
  COLOR_RGB: true
  DATASET: 'coco'
  DATA_FORMAT: jpg
  FLIP: true
  NUM_JOINTS_HALF_BODY: 8
  PROB_HALF_BODY: 0.3
  ROOT: 'data/coco/'
  ROT_FACTOR: 45
  SCALE_FACTOR: 0.35
  TEST_SET: 'val2017'
  TRAIN_SET: 'train2017'
```
and a longer training schedule (203 epochs), to train the original SimpleBaseline-Res50+DarkPose, it now brings 0.4 AP gain (72.0->72.4) on coco val datset with the same detected boxes (56.4AP). 

| Arch | AP | Ap .5 | AP .75 | AP (M) | AP (L) | AR | AR .5 | AR .75 | AR (M) | AR (L) |
|---|---|---|---|---|---|---|---|---|---|---|
| pose_resnet | 0.724 | 0.894 | 0.796 | 0.685 | 0.795 | 0.779 | 0.932 | 0.841 | 0.734 | 0.844 |

**Comment by yangsenius** [comment]
Hi, @HuangJunJie2017 . We did not train our models using the aug-schedule configuration of the original SimpleBaseline.

We provide the convergence curves and [log](https://gist.github.com/yangsenius/f6aaad3ad2eab37f39afa4b39924f7a0#file-gistfile1-txt) of the mAP for TransPose-R-A4. The curves are as below:

![Screenshot from 2021-01-27 10-54-49](https://user-images.githubusercontent.com/28689106/105942616-8656d600-609a-11eb-87bc-9842896e8603.png)

We also use the same data augmentation strategy of TransPose-R, 
```yaml
DATASET:
  COLOR_RGB: true
  DATASET: 'coco'
  DATA_FORMAT: jpg
  FLIP: true
  NUM_JOINTS_HALF_BODY: 8
  PROB_HALF_BODY: 0.3
  ROOT: 'data/coco/'
  ROT_FACTOR: 45
  SCALE_FACTOR: 0.35
  TEST_SET: 'val2017'
  TRAIN_SET: 'train2017'
```
and a longer training schedule (203 epochs), to train the original SimpleBaseline-Res50+DarkPose, it now brings 0.4 AP gain (72.0->72.4) on coco val datset with the same detected boxes (56.4AP). 

| Arch | AP | Ap .5 | AP .75 | AP (M) | AP (L) | AR | AR .5 | AR .75 | AR (M) | AR (L) |
|---|---|---|---|---|---|---|---|---|---|---|
| pose_resnet | 0.724 | 0.894 | 0.796 | 0.685 | 0.795 | 0.779 | 0.932 | 0.841 | 0.734 | 0.844 |

**Comment by HuangJunJie2017** [comment]
emmm, i have not try SimpleBaseline-Res50+darkpose+hrnetaugschedule. But i have tried another similar configuration SimpleBaseline-Res50+UDP+hrnetaugschedule. This configuration scores 73.4 AP on coco val datset with the same detected boxes (56.4AP).  72.0 for SimpleBaseline-Res50+darkpose+hrnetaugschedule is too low. There may be some problem in your experiment which may has negative impact on both  SimpleBaseline and  TransPose? TransPose can perform better?

**Comment by yangsenius** [comment]
We run the official DarkPose code on our machine. Here we provide the history log files.

[orignal_config_baseline_res50_256x192_d256x3_adam_lr1e-3_2020-11-10-13-01_train.log](https://github.com/yangsenius/TransPose/files/5878102/orignal_config_baseline_res50_256x192_d256x3_adam_lr1e-3_2020-11-10-13-01_train.log)

[orignal_config_baseline_res50_256x192_d256x3_adam_lr1e-3_2020-11-11-20-52_valid.log](https://github.com/yangsenius/TransPose/files/5878115/orignal_config_baseline_res50_256x192_d256x3_adam_lr1e-3_2020-11-11-20-52_valid.log)

**Comment by HuangJunJie2017** [comment]
emm, it it seem that you don't train SimpleBaseline-Res50+darkpose+hrnetaugschedule this with 210 epoch, but 140 epoch instead. This make this configuration perform poor (72.0) ~, if you train this with 210 epoch, SimpleBaseline-Res50+darkpose+hrnetaugschedule will score more than 73.0 AP coco val datset with the same detected boxes (56.4AP).

**Comment by yangsenius** [comment]
Yes, a longer scheduler or stronger augmentation indeed increases the AP. But, the original SimpleBaseline-Res50 and SimpleBaseline-Res50+DarkPose are trained with 140 epochs. Due to we use the data augmentation of HRNet for TransPose-R, so we trained SimpleBaseline-Res50+DarkPose with stronger data augmentation for a fair comparison. In addition, the current result I reported (204 epochs) is 72.4 AP now. Maybe a better result will be achieved by addining training steps.
